### PR TITLE
Update GitHub CI to use built-in GITHUB token & fix bad test

### PR DESCRIPTION
**Pull request by er1c**

### CODE REVIEW

File: core/src/main/scala/repatch/github/response/Paged.scala [hunk 1]
```diff
@@ -30,10 +30,10 @@ object Paged extends Parse {
     val json = as.json4s.Json(res)
     val links = linkHeader(res)
     Paged(
-      for {
-        JArray(array) <- json
-        v <- array
-      } yield f(v),
+      json match {
```
Comment (er1c) [inline core/src/main/scala/repatch/github/response/Paged.scala:33]: this is the real fix, other stuff was just crap to identify what was failing

File: core/src/test/scala/repatch/github/GithubSpec.scala [hunk 2]
```diff
@@ -49,9 +79,6 @@ class GithubSpec extends Specification {
   `gh.repo(:owner, :repo).git_blob(:sha).raw` should
     return raw blob bytes                                                     ${raw1}
 
-  `gh.issues` should
```
Comment (er1c) [inline core/src/test/scala/repatch/github/GithubSpec.scala:52]: and these removals

File: core/src/test/scala/repatch/github/GithubSpec.scala [hunk 3]
```diff
@@ -154,10 +175,15 @@ class GithubSpec extends Specification {
   }
 
   def repos4 = {
-    val repos = http(client(gh.user.repos.asc) > as.repatch.github.response.Repos)
+    val repos = http(client(gh.user(user).repos.asc) > as.repatch.github.response.Repos)
     repos().head.full_name must_!= "foo"
   }
 
+  def users1 = {
```
Comment (er1c) [inline core/src/test/scala/repatch/github/GithubSpec.scala]: this isn't used

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -16,6 +16,7 @@ jobs:
             java: 11
             jobtype: 1
     runs-on: ${{ matrix.os }}
+    permissions: read-all
```
Comment (er1c) [inline .github/workflows/ci.yml]: let me double check if there is a more restrictive that works

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**